srcalc/src/main/java/gov/va/med/srcalc/web/controller/EnterVariablesController.java

```diff
@@ -88,6 +90,36 @@ public ModelAndView enterVariables(
             parserVisitor.visit(variable);
         }
         
+        try
+        {
+        	// Set the values on the Calculation.
+        	fCalculationService.runCalculation(
+        			workflow.getCalculation(), parserVisitor.getValues());
+        }
+        catch(final MissingValueListException e)
+        {
+        	// Add all of the missing values to the binding errors
+        	for(final MissingValueException missingValue : e.getMissingValues())
+        	{
+        		// If the variable had invalid input, rather than missing input,
+        		// it would never be included in the values.
+        		fLogger.debug("Field Error: {}", VariableEntry.makeDynamicValuePath(missingValue.getVariable().getKey()));
+        		final String dynamicKey = VariableEntry.makeDynamicValuePath(missingValue.getVariable().getKey());
+        		// Account for field errors on special fields like discrete numerical
+        		// variables.
+        		final String numericalKey = VariableEntry.makeDynamicValuePath(
+        				missingValue.getVariable().getKey() + VariableEntry.SEPARATOR + VariableEntry.SPECIAL_NUMERICAL);
+        		if(!valuesBindingResult.hasFieldErrors(dynamicKey) &&
```

Comment:
It would be better to encapsulate these two tests in a method like `bindingResultAlreadyHasError(dynamicKey)`.